packages/truffle-db/src/workspace/test/utils.ts

```diff
@@ -16,11 +17,16 @@ export const fixturesDirectory = path.join(
   "fixtures"
 );
 
+const tempDir = tmp.dirSync({ unsafeCleanup: true });
+tmp.setGracefulCleanup();
+
 export class WorkspaceClient {
   private workspace: Workspace;
+  private persistedWorkspace: Workspace;
 
   constructor () {
-    this.workspace = new Workspace();
+    this.workspace = new Workspace(tempDir.name);
+    this.persistedWorkspace = new Workspace(tempDir.name);
```

Comment:
Are you sure you want to create two workspaces all the time? Maybe this could be a `constructor` argument (cause we really only need the second workspace for that particular test)

Reply:
I now only create the persisted workspace in the execute function if the appropriate parameter is passed. This way it is only done for the persistence test!